edb/lang/graphql/translator.py

```diff
@@ -459,12 +464,68 @@ def _validate_fragment_type(self, frag, spread):
     def _visit_arguments(self, arguments):
         where = offset = limit = None
         orderby = []
+        first = last = before = after = None
 
         for arg in arguments:
-            if arg.name == 'filter':
-                where = self.visit(arg.value)
-            elif arg.name == 'order':
-                orderby = self.visit_order(arg.value)
+            try:
+                if arg.name == 'filter':
+                    where = self.visit(arg.value)
+                elif arg.name == 'order':
+                    orderby = self.visit_order(arg.value)
+                elif arg.name == 'first':
+                    first = arg.value.value
+                    if first < 0:
+                        raise ValueError(f"'{arg.name}' cannot be negative")
```

Comment:
`raise ValueError(f"'{arg.name}' cannot be negative")` -> raise ValueError(f"{arg.name!r} cannot be negative")

Reply:
OK.